src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -1030,646 +1301,847 @@ public enum SupremeVictors implements LogicCardInfo {
-            text "Choose 1 of your opponent’s Pokémon that has any damage counters on it. This attack does 30 damage to that Pokémon."
+            text "Choose 1 of your opponent's Pokémon that has any damage counters on it. This attack does 30 damage to that Pokémon. (Don't apply Weakness and Resistance for Benched Pokémon.)"
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert opp.all.find{it.numberOfDamageCounters} : "Your opponent's Pokémon are healthy"
+            }
             onAttack {
-              damage 0
+              damage 30, opp.all.findAll{it.numberOfDamageCounters}.select(text)
             }
           }
-
         };
       case SOLROCK_45:
-        return basic (this, hp:HP080, type:PSYCHIC, retreatCost:1) {
+        return basic (this, hp:HP080, type:P, retreatCost:1) {
           weakness G, PLUS20
           pokePower "Sunshine Fate", {
-            text "Once during your turn , if you have Lunatone in play, you may look at the top 3 cards of your deck and put them back on top of your deck in any order. This power can’t be used if Solrock is affected by a Special Condition."
+            text "Once during your turn (before your attack), if you have Lunatone in play, you may look at the top 3 cards of your deck and put them back on top of your deck in any order. This power can't be used if Solrock is affected by a Special Condition."
             actionA {
+              checkLastTurn()
+              checkNoSPC()
+              assert my.all.find{it.name == "Lunatone"} : "You don't have Lunatone in play"
+              assert my.deck : "Your deck is empty"
+              powerUsed()
+              def maxSize = Math.min(my.deck.size(),3)
+              def list = rearrange(my.deck.subList(0,maxSize), "Rearrange top $maxSize cards of your deck")
+              my.deck.setSubList(0, list)
+              bc "$thisAbility rearranged the top cards of ${self.owner.getPlayerUsername(bg)}'s deck"
             }
           }
           move "Luna Turn", {
             text "30+ damage. If Lunatone is in your discard pile, this attack does 30 damage plus 30 more damage. Then, search your discard pile for Lunatone, show it to your opponent, and shuffle it into your deck."
             energyCost C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              if(my.discard.find{it.name == "Lunatone"}) {
+                damage 30
+                afterDamage {
+                  my.discard.select("Search your discard pile for Lunatone",{it.name == "Lunatone"}).showToOpponent("Luna Turn: Selected cards").moveTo(my.deck)
+                  shuffleDeck()
+                }
+              }
             }
           }
-
         };
       case SPINDA_46:
-        return basic (this, hp:HP070, type:COLORLESS, retreatCost:1) {
+        return basic (this, hp:HP070, type:C, retreatCost:1) {
```

Comment:
This was the reason for adding colorless energy to all of the rainbow energy cards. The logic is that it iterates through all types and if the type is shared among energy cards on both spinda and the defending pokemon, it adds every energy that has that type to the list of all energy cards that could be chosen to be discarded.

Reply:
I'm aware, I figure it was tested enough to not need to review it, but was mentioning that I was explicitly skipping it just in case it could be relevant in the future :P

Reply:
Fair enough. Sorry I misunderstood your intent.